coderd/healthcheck/accessurl.go

```diff
@@ -0,0 +1,48 @@
+package healthcheck
+
+import (
+	"context"
+	"io"
+	"net/http"
+	"net/url"
+
+	"golang.org/x/xerrors"
+)
+
+type AccessURLReport struct {
+	Reachable       bool
+	StatusCode      int
+	HealthzResponse string
+	Err             error
+}
+
+func (r *AccessURLReport) Run(ctx context.Context, accessURL *url.URL) {
+	accessURL, err := accessURL.Parse("/healthz")
+	if err != nil {
+		r.Err = xerrors.Errorf("parse healthz endpoint: %w", err)
+		return
+	}
+
+	req, err := http.NewRequestWithContext(ctx, "GET", accessURL.String(), nil)
```

Comment:
Even though the healthcheck has a 30 second context, ideally the access URL healthcheck should fail if it takes more than 5 seconds or so.

Reply:
I forgot I wrote this, gonna leave out of this pr